Bookmark-Manager/src/test/java/com/bookmark/bookmarks/BookmarkServiceTest.java

```diff
@@ -0,0 +1,215 @@
+package com.bookmark.bookmarks;
+
+import com.bookmarkmanager.bookmark.BookmarkRepository;
+import com.bookmarkmanager.bookmark.BookmarkService;
+import com.bookmarkmanager.bookmarkfolder.FolderRepository;
+import com.bookmarkmanager.exception.BookmarkManagerException;
+import com.bookmarkmanager.exception.BookmarkNotFoundException;
+import com.bookmarkmanager.exception.ResourceNotFoundException;
+import com.bookmarkmanager.pojo.Bookmark;
+import com.bookmarkmanager.pojo.Folder;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+class BookmarkServiceTest {
+
+  BookmarkRepository bookmarkRepository = mock(BookmarkRepository.class);
+  FolderRepository folderRepository = mock(FolderRepository.class);
+  BookmarkService bookmarkService = new BookmarkService(bookmarkRepository, folderRepository);
+
+
+
+  @Test
+  void test_create_bookmark_with_valid_data() {
+    UUID folderId = UUID.randomUUID();
+    Bookmark bookmark = new Bookmark();
+    bookmark.setTitle("Sample Title");
+    bookmark.setUrl("http://example.com");
+    bookmark.setUserId(UUID.randomUUID());
+    bookmark.setFolderId(folderId);
+
+    when(folderRepository.findById(folderId)).thenReturn(Optional.of(new Folder()));
+    when(bookmarkRepository.save(any(Bookmark.class))).thenReturn(bookmark);
+
+    Bookmark savedBookmark = bookmarkService.createBookmark(bookmark);
+
+    assertNotNull(savedBookmark);
+    assertEquals("Sample Title", savedBookmark.getTitle());
+    assertEquals("http://example.com", savedBookmark.getUrl());
+  }
+
+    @Test
+    void test_create_bookmark_with_invalid_data() {
```

Comment:
Missing Tests
Consider adding tests for:
- Boundary conditions for URL and title lengths
- Special characters in URLs
- Duplicate bookmark detection (if applicable)
- Case sensitivity handling